### PR TITLE
Psy maps update to resolve #34

### LINKED ISSUE #34: upgrade map plot gridded (lat/lon) netCDF data (psyplot)
- migrate to psyplot 1.2.1 

- Add a field to allow users to specify the title for their plot

- When netCDF file contains more than one field, it plots the first one only.
     - Add an option to select one index when the file contains more than one field (could also be via a variable value such as time/year, etc.)
     - Add an option to have multiple plots on the same plot.

### PR DESCRIPTION
**Pull request by annefou**
Can now generate multiple plot when netCDF file contains several fields (time is used for selection)

### CODE REVIEW

File: tools/psy-maps/psymap_simple.py [hunk 1]
```diff
@@ -49,37 +72,110 @@ def __init__(self, input, proj, varname, cmap, output,
...
             print("cmap: ", self.cmap)
+            print("time: ", self.time)
+            print("ncol: ", self.ncol)
+            print("nrow: ", self.nrow)
+            print("title: ", self.title)
+            print("date format: ", self.format)
             print("output: ", self.output)
 
     def plot(self):
+        if self.title == "" and self.format == "":
```
Comment (bgruening) [inline tools/psy-maps/psymap_simple.py]: `if self.title and self.format`

File: tools/psy-maps/psymap_simple.py [hunk 2]
```diff
@@ -49,37 +72,110 @@ def __init__(self, input, proj, varname, cmap, output,
...
+        else:
+            title = self.title + "\n" + self.format
+
         if self.cmap is None and self.proj is None:
             print("op1")
             psy.plot.mapplot(self.input, name=self.varname,
+                             title=title,
                              clabel='{desc}')
-        elif self.proj is None or self.proj == '':
+        elif (self.proj is None or self.proj == ''):
```
Comment (bgruening) [inline tools/psy-maps/psymap_simple.py]: the brackets are not needed, do you need the print one below?

File: tools/psy-maps/psymap_simple.py (hunk 1 above)
Comment (annefou) [reply tools/psy-maps/psymap_simple.py]: Thanks! I updated it (in all the code).

File: tools/psy-maps/psymap_simple.py (hunk 2 above)
Comment (annefou) [reply tools/psy-maps/psymap_simple.py]: Removed brackets and prints.

File: tools/psy-maps/psymap_simple.py [hunk 3]
```diff
@@ -49,37 +72,110 @@ def __init__(self, input, proj, varname, cmap, output,
...
+
+        if self.title == "":
+            title = self.format
+        else:
+            title = self.title + "\n" + self.format
+        mpl.rcParams['figure.figsize'] = [20, 8]
+        mpl.rcParams.update({'font.size': 8})
+        rcParams.update({'plotter.maps.grid_labelsize': 8.0})
+        if self.cmap is None and self.proj is None:
+            print("op1 time")
```
[the remark this answers is not in the document]
Comment (annefou) [reply tools/psy-maps/psymap_simple.py]: Good point. I forgot to remove them! Thanks a lot!

File: tools/psy-maps/psy-maps.xml [hunk 4]
```diff
@@ -1,23 +1,49 @@
...
             --proj '$adv.projection'
             --cmap '$adv.colormap'
             --output image.png
+       #if str($adv.multiple_plot.selector) == 'yes'
+            --time '$adv.multiple_plot.time'
+            --format '$adv.multiple_plot.format'
+            --ncol '$adv.multiple_plot.ncol'
+            --nrow '$adv.multiple_plot.nrow'
+       #end if
+       #if str($adv.multiple_plot.title) != ""
```
Comment (bgruening) [inline tools/psy-maps/psy-maps.xml]:
```suggestion
       #if str($adv.multiple_plot.title).strip()
```

File: tools/psy-maps/psy-maps.xml [hunk 5]
```diff
@@ -1,23 +1,49 @@
...
         <param name="variable" type="text" value="TS" label="variable name as given in the netCDF file" />
         <section name="adv" title="Advanced Options" expanded="false">
+            <conditional name="multiple_plot">
+                <param name="selector" type="select" label="multiple times">
+                    <option value="yes">yes</option>
+                    <option value="no" selected="true">no</option>
+                </param>
+                <when value="yes">
+                    <param name="time" type="text" value="0" label="comma separated list of indexes for fields to plot" />
+                    <param name="format" type="text" value="%Y-%m-%d %H:%M:%S" label="date and time format" />
```
Comment (bgruening) [inline tools/psy-maps/psy-maps.xml]: we should really get a data and time picker ...

Comment (annefou) [reply tools/psy-maps/psy-maps.xml]: Yes! I still struggle to find a good way to handle time in Galaxy. I always wonder if we need to have a date and time datatype too. I think I need to develop some more tools to get more experience! 

Thanks for reviewing Climate tools!

Comment (bgruening) [reply tools/psy-maps/psy-maps.xml]: https://github.com/galaxyproject/galaxy/issues/6806 as reference for later.

Comment (bgruening) [reply tools/psy-maps/psy-maps.xml]: https://github.com/galaxyproject/galaxy/issues/6806 as reference for later.

**Pull request merged**